sympy/core/basic.py

```diff
@@ -1557,12 +1557,24 @@ def doit(self, **hints):
             return self
 
     def _eval_rewrite(self, pattern, rule, **hints):
+        from sympy.simplify.simplify import simplify
         if self.is_Atom:
             if hasattr(self, rule):
                 return getattr(self, rule)()
             return self
 
-        if hints.get('deep', True):
+        if rule == '_eval_rewrite_as_sinc':
```

Comment:
I don't like this hack.  I also don't like calling full simplify in a rewrite routine. What happens if you don't make this change here? I'm unclear what the problem is.